src/state/slices/color_ramp_slice.ts

```diff
@@ -215,32 +226,45 @@ export const addColorRampDerivedStateSubscribers = (
   );
 };
 
-export const serializeColorRampSlice = (slice: ColorRampSlice): SerializedStoreData => {
+export const serializeColorRampSlice = (slice: Partial<ColorRampSliceSerializableState>): SerializedStoreData => {
   const ret: SerializedStoreData = {};
 
   // Ramp + reversed
-  ret[UrlParam.COLOR_RAMP] =
-    slice.colorRampKey + (slice.isColorRampReversed ? UrlParam.COLOR_RAMP_REVERSED_SUFFIX : "");
+  if (slice.colorRampKey !== undefined) {
+    ret[UrlParam.COLOR_RAMP] = slice.colorRampKey + (slice.isColorRampReversed ? URL_COLOR_RAMP_REVERSED_SUFFIX : "");
+  }
 
-  ret[UrlParam.KEEP_RANGE] = encodeBoolean(slice.keepColorRampRange);
+  ret[UrlParam.KEEP_RANGE] = encodeMaybeBoolean(slice.keepColorRampRange);
 
-  const range = slice.colorRampRange;
-  ret[UrlParam.RANGE] = range.map(encodeNumber).join(",");
+  if (slice.colorRampRange !== undefined) {
+    const range = slice.colorRampRange;
+    ret[UrlParam.RANGE] = range.map(encodeNumber).join(",");
+  }
```

Comment:
I recommend turning off whitespace changes, since some of these changes are just me adding checks for `undefined` values.